### PR TITLE
Kafka Connect Avro Support Part 2: AvroEventBatchEncoder

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?
#660 

### What is changed and how it works?
Add AvroEventBatchEncoder which handles RowChangedEvent and DDLEvent.

### Check List 

Tests 

 - Unit test
 - Integration test: will be done in the next stage after the sink is ready.

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

TODO: 
 - Use primary keys as Kafka key.

### Release note
- No release note

### CODE REVIEW

File: cdc/sink/codec/avro.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+		return "null"
+	case string:
+		return "string"
+	case time.Duration:
+		return "long.time-millis"
+	case time.Time:
+		return "long.timestamp-millis"
+	default:
+		log.Warn("getAvroDataTypeName: unknown type")
+		return "errorType"
```
Comment (overvenus) [inline cdc/sink/codec/avro.go:230]: Is "errorType" valid in Avro?

File: cdc/sink/codec/avro.go [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+	evlp, err := res.toEnvelope()
+	if err != nil {
+		log.Warn("AppendRowChangedEvent: could not construct Avro envelope", zap.String("table", e.Table.String()))
+		return errors.Annotate(err, "AppendRowChangedEvent could not construct Avro envelope")
+	}
+
+	a.valueBuf = evlp
+	// TODO use primary key(s) as kafka key
+	a.keyBuf = []byte(fmt.Sprintf("%d", e.RowID))
```
Comment (amyangfei) [inline cdc/sink/codec/avro.go:80]: Use `strconv.FormatInt` is faster than `fmt.Sprintf`

File: cdc/sink/codec/avro.go [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+		return "string"
+	case mysql.TypeTinyBlob, mysql.TypeMediumBlob, mysql.TypeLongBlob, mysql.TypeBlob:
+		return "bytes"
+	default:
+		return "error"
+	}
+}
+
+func columnToAvroNativeData(col *model.Column) (interface{}, string, error) {
+	if v, e := col.Value.(int); e {
```
Comment (amyangfei) [inline cdc/sink/codec/avro.go:270]:
```suggestion
	if v, ok := col.Value.(int); ok {
```
`ok` is used more often in type assertion in golang.

File: cdc/model/sink.go [hunk 4]
```diff
@@ -82,14 +83,28 @@ type Column struct {
...
-	Schema   string
-	Table    string
-	Query    string
-	Type     model.ActionType
+	StartTs    uint64
+	CommitTs   uint64
+	Schema     string
+	SchemaID   int64
+	Table      string
+	ColumnInfo []*ColumnInfo
```
Comment (leoppro) [inline cdc/model/sink.go:105]: I think we could use a `TableInfo` with a table name, schema name, column info, etc.

**Comment by amyangfei** [comment]
Please fix the build error @liuzix

**Pull request merged**